### PR TITLE
Dynamic graph version of  a2c in Paddle

### PR DESCRIPTION
**Pull request by wangzelong0663**
Add the dynamic graph version of  a2c with Paddle.

### CODE REVIEW

File: benchmark/paddle/A2C/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+## Reproduce A2C with PARL
+Based on PARL, the A2C algorithm of deep reinforcement learning has been reproduced, reaching the same level of indicators as the paper in Atari benchmarks.
+
+A2C is a synchronous, deterministic variant of [Asynchronous Advantage Actor Critic (A3C)](https://arxiv.org/abs/1602.01783). Instead of updating asynchronously in A3C or GA3C, A2C uses a synchronous approach that waits for each actor to finish its sampling before performing an update. We use a common a2c algotrithm `parl.algorithms.A2C`(paddle version)for the A2C examples.
```
Comment (TomorrowIsAnOtherDay) [inline benchmark/paddle/A2C/README.md]: Remove the last line.

File: benchmark/paddle/A2C/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+### Atari games introduction
```
Comment (TomorrowIsAnOtherDay) [inline benchmark/paddle/A2C/README.md]: games -> game

File: benchmark/paddle/A2C/README.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+### Dependencies
++ [paddlepaddle>=2.0.0](https://github.com/PaddlePaddle/Paddle)
++ [parl>=1.4.3](https://github.com/PaddlePaddle/PARL)
++ gym==0.12.1
++ atari-py==0.1.7
+
+
+### Distributed Training
+
+At first, We can start a local cluster with 5 CPUs:
```
Comment (TomorrowIsAnOtherDay) [inline benchmark/paddle/A2C/README.md]: We -> we

File: benchmark/paddle/A2C/a2c_config.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
```
Comment (TomorrowIsAnOtherDay) [inline benchmark/paddle/A2C/a2c_config.py]: 2018 -> 2021. Same problem for other files.

File: benchmark/paddle/A2C/atari_model.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+            out_channels=64,
+            kernel_size=3,
+            stride=1,
+            padding=0)
+
+        self.flatten = nn.Flatten()
+
+        # Need to calc the size of the in_features according to the input image.
+        # The default size of the input image is 84 * 84
+        self.fc = nn.Linear(in_features=64 * 9 * 9, out_features=512)
```
Comment (TomorrowIsAnOtherDay) [inline benchmark/paddle/A2C/atari_model.py:43]: In torch-based implementation, the number of neurons for the first fc layer is  7744. https://github.com/PaddlePaddle/PARL/blob/ab1eb893a9f0ac1d5238c6a5277ea7e7c6cd1fdf/benchmark/torch/a2c/atari_model.py#L40
What causes inequality here?

File: benchmark/paddle/A2C/train.py [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+        obs_shape = env.observation_space.shape
+        act_dim = env.action_space.n
+        self.config['obs_shape'] = obs_shape
+        self.config['act_dim'] = act_dim
+
+        model = AtariModel(act_dim)
+        algorithm = A2C(model, vf_loss_coeff=config['vf_loss_coeff'])
+        self.agent = AtariAgent(algorithm, config)
+
+        if machine_info.is_gpu_available():
```
Comment (TomorrowIsAnOtherDay) [inline benchmark/paddle/A2C/train.py]: Can we remove line 51~53? @zenghsh3

File: benchmark/paddle/A2C/atari_model.py (hunk 5 above)
Comment (wangzelong0663) [reply benchmark/paddle/A2C/atari_model.py:43]: The model architecture is same to the paddle fluid version.
https://github.com/PaddlePaddle/PARL/blob/ab1eb893a9f0ac1d5238c6a5277ea7e7c6cd1fdf/examples/A2C/atari_model.py#L23-L30
And the model architecture of the torch version is:
https://github.com/PaddlePaddle/PARL/blob/ab1eb893a9f0ac1d5238c6a5277ea7e7c6cd1fdf/benchmark/torch/a2c/atari_model.py#L26-L40
The padding is different...
Maybe the model architecture of the torch-based implementation is needed to modify.

Comment (TomorrowIsAnOtherDay) [reply benchmark/paddle/A2C/atari_model.py:43]: I see. We can update the architecture of torch implementation later.

File: benchmark/paddle/A2C/train.py (hunk 6 above)
Comment (zenghsh3) [reply benchmark/paddle/A2C/train.py]: It's unnecessary in the paddle dynamic version.

File: parl/algorithms/paddle/a2c.py [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+            actions: An int64 numpy array of shape [B].
+            advantages: A float32 numpy array of shape [B].
+            target_values: A float32 tensor of shape [B].
+            learning_rate: float scalar of leanring rate.
+            entropy_coeff: float scalar of entropy coefficient.
+        """
+        # shape: [B, act_dim]
+        logits = self.model.policy(obs)
+        act_dim = logits.shape[-1]
+        actions = paddle.to_tensor(actions)
```
Comment (zenghsh3) [inline parl/algorithms/paddle/a2c.py]: Put the logic of `numpy  to tensor` and `tensor to numpy` in agent.

**Review by zenghsh3 — APPROVED** [review]
LGTM
